presto-native-execution/presto_cpp/main/benchmarks/http-jwt/HttpJwtBenchmarkTest.cpp

```diff
@@ -0,0 +1,129 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include "presto_cpp/main/http/filters/JwtInternalAuthFilter.h"
+#include "presto_cpp/main/http/tests/HttpTest.h"
+
+namespace {
+
+class HttpJwtBenchmark : public ::testing::TestWithParam<bool> {
+ public:
+  void TestBody(){};
+
+  void nojwtRun() {
+    folly::BenchmarkSuspender suspender;
+    auto memoryPool = defaultMemoryManager().addLeafPool("basicJwtTest");
+
+    // Don't use https
+    auto server = getServer(false);
+    server->registerGet("/ping", ping);
+    auto wrapper = std::make_unique<HttpServerWrapper>(std::move(server));
+    auto serverAddress = wrapper->start().get();
+
+    HttpClientFactory clientFactory;
+    auto client = clientFactory.newClient(
+        serverAddress, std::chrono::milliseconds(1'000), false, memoryPool);
+    suspender.dismiss();
+
+    auto response = sendGet(client.get(), "/ping").get();
+
+    suspender.rehire();
+    wrapper->stop();
+    suspender.dismiss();
+  }
+
+  void jwtRun() {
+    folly::BenchmarkSuspender suspender;
+    auto memoryPool = defaultMemoryManager().addLeafPool("basicJwtTest");
+
+    const std::string kSecret{"mysecret"};
+    const std::string kNodeId{"testnode"};
+
+    std::unordered_map<std::string, std::string> systemValues{
+        {std::string(SystemConfig::kMutableConfig), std::string("true")},
+        {std::string(SystemConfig::kInternalCommunicationJwtEnabled),
+         std::string("true")},
+        {std::string(SystemConfig::kInternalCommunicationSharedSecret),
+         kSecret}};
+    std::unique_ptr<Config> rawSystemConfig =
+        std::make_unique<core::MemConfigMutable>(systemValues);
+
+    auto systemConfig = SystemConfig::instance();
+    systemConfig->initialize(std::move(rawSystemConfig));
+
+    std::unordered_map<std::string, std::string> nodeValues{
+        {std::string(NodeConfig::kMutableConfig), std::string("true")},
+        {std::string(NodeConfig::kNodeId), std::string(kNodeId)}};
+    std::unique_ptr<Config> rawNodeConfig =
+        std::make_unique<core::MemConfigMutable>(nodeValues);
+    auto nodeConfig = NodeConfig::instance();
+    nodeConfig->initialize(std::move(rawNodeConfig));
+
+    // Don't use https
+    auto server = getServer(false);
+
+    server->registerGet("/ping", ping);
+
+#ifdef PRESTO_ENABLE_JWT_TOKEN
+    std::vector<std::unique_ptr<proxygen::RequestHandlerFactory>> filters;
+    filters.push_back(std::make_unique<http::filters::JWTTokenFilterFactory>());
+#endif
+    auto wrapper = std::make_unique<HttpServerWrapper>(std::move(server));
+#ifdef PRESTO_ENABLE_JWT_TOKEN
```

Comment:
The ifdefs are not needed anymore. The benchmark is only built if both benchmarks and presto_jwt_enable is set. I removed them and also changed the order that would allow a single ifdef block.